platforms/web/lib/composer.ts

```diff
@@ -71,27 +72,38 @@ export function processInput(
     }
 
     switch (event.inputType) {
+        case 'insertAtRoomSuggestion': {
+            if (suggestion && isAtRoomSuggestionEvent(event)) {
+                const { attributes } = event.data;
+                // we need to track data-mention-type in element web, ensure we do not pass
+                // it in as rust model can handle this automatically
+                if (attributes.has('data-mention-type')) {
+                    attributes.delete('data-mention-type');
+                }
+                return action(
+                    composerModel.insert_at_room_mention_at_suggestion(
+                        suggestion,
+                        attributes,
+                    ),
+                    'insert_at_room_mention_at_suggestion',
+                );
+            }
+            break;
+        }
```

Comment:
This replaces the `if text === ...` check from before to completely separate mention and at mention handling